mcpy/mcsimulation.py

```diff
@@ -195,10 +195,9 @@ def calculate_total_energy(self):
         be done at initialization though can be called at any time.
         '''
         e_total = 0
-        for i in np.arange(self.particles.num_particles):
-            rij2 = self.box.minimum_image_distance(
-                self.particles.coordinates[i],
-                self.particles.coordinates[i+1:]
+        for i in np.arange(self.particles.num_particles - 1):
+            rij2 = self.box.minimum_image_distance(0,
+                    self.particles.coordinates[i:]
```

Comment:
The fixed 'index' value being 0 may lead to a problem. I tested a generic n x 3 array in python and it will yield:
> coord[:0,:]
array([ ], shape=(0, 3), dtype=int64)

May you double check that?